scripts/source-maintenance.sh

```diff
@@ -84,27 +100,53 @@ done
 # an error code seen by a KeepGoing-aware command (or zero)
 SeenErrors=0
 
-
 if ! git diff --quiet; then
 	echo "There are unstaged changes. This script may modify sources."
 	echo "Stage changes to avoid permanent losses when things go bad."
 	exit 1
 fi
 
-# On squid-cache.org we have to use the python scripted md5sum
-HOST=`hostname`
-if test "$HOST" = "squid-cache.org" ; then
-	MD5="md5"
+for Checksum in md5sum md5 shasum sha1sum none
```

Comment:
I chose "none" with autoconf conventions in mind but it's fine to change it

Reply:
"None" is a valid word in many contexts, but I doubt autoconf runs a program called "none" in hope that its execution fails without bad side effects.

Reply:
```
kinkie@n2:~/src/squid$ grep none configure.ac
AC_PATH_PROG(PERL, perl, none)
if test "x$ac_cv_path_PERL" = "xnone"; then
  AC_PATH_PROG(WIN32_PSAPI, psapi.dll, none)
  if test "x$ac_cv_path_WIN32_PSAPI" = "xnone"; then
AC_MSG_RESULT([${squid_disk_module_candidates:-none}])
for module in $squid_disk_module_candidates none; do
  if test "x$module" = "xnone"; then
for fs in $squid_storeio_module_candidates none; do
  elif test "x$ac_cv_search_epoll_ctl" = "xnone required" ; then
if test "x$enableval" != "xnone" -a "x$enableval" != "xno" ; then
  squid_opt_default_hosts="none"
      To enable but build no helpers, specify "none".
      To enable but build no helpers, specify "none".
      To enable but build no helpers, specify "none".
      To enable but build no helpers, specify "none".
                  To build no helpers, specify "none".
                  helpers. Use --disable-url-rewrite-helpers to build none.
                  build none. For a list of available helpers see the
                  build none. For a list of available helpers see the
                  helpers. Use --disable-storeid-rewrite-helpers to build none.
```

Reply:
In the configure.ac context you pasted, "none" is not a name of the program to run and fail, but a special value that means "not found". In AC_PATH_PROG terminology: not `prog-to-check-for` but `value-if-not-found`.

In this PR, you have worked on two loops for finding programs. One loop runs a special program that should fail. The other one does not -- it detects the case where all attempts failed without resorting to an always-failing trailing attempt. The latter approach is a lot cleaner (and can be argued to resemble the autoconf approach, just with an empty string instead of "none" as a special value).